core/worker.go

```diff
@@ -1178,6 +1178,31 @@ func (w *worker) prepareWork(genParams *generateParams, wo *types.WorkObject) (*
 		newWo.Header().SetInterlinkRootHash(interlinkRootHash)
 	}
 
+	if nodeCtx == common.PRIME_CTX {
+		if w.hc.IsGenesisHash(parent.Hash()) {
+			newWo.Header().SetQiToQuai(big.NewInt(0))
+			newWo.Header().SetQuaiToQi(big.NewInt(0))
+			newWo.Header().SetExchangeRate(parent.Header().ExchangeRate())
+		} else {
+			newQiToQuai, newQuaiToQi, newExchangeRate, err := misc.CalculateExchangeRate(parent)
+			if err != nil {
+				newWo.Header().SetQiToQuai(newQiToQuai)
+				newWo.Header().SetQuaiToQi(newQuaiToQi)
+				newWo.Header().SetExchangeRate(newExchangeRate)
+			}
+		}
+	} else {
+		if w.hc.IsGenesisHash(parent.Hash()) {
+			newWo.Header().SetQiToQuai(big.NewInt(0))
```

Comment:
not sure if we need this, as the magic of combining happens in the slice.go combinependingheader